static/openapi/faas-spec.json

```diff
@@ -0,0 +1,670 @@
+{
+  "openapi": "3.0.3",
+  "info": {
+    "title": "PhotonIQ FaaS API",
+    "description": "API documentation for the PhotonIQ FaaS Service",
+    "version": "1.0.0"
+  },
+  "servers": [
+    {
+      "url": "https://api-demo-faas.paas.macrometa.io",
+      "description": "PhotonIQ FaaS API"
+    }
+  ],
+  "paths": {
+    "/api/faas/v1/health": {
+      "get": {
+        "tags": [
+          "Health"
+        ],
+        "summary": "Check system health",
+        "description": "Check if system is up and running.",
+        "operationId": "handle_mm_health",
+        "responses": {
+          "200": {
+            "description": "The system is up and running."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/deploy": {
+      "post": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Deploy the function",
+        "description": "Deploy the function for the specified key function name.",
+        "operationId": "handle_mm_deploy",
+        "parameters": [
+          {
+            "name": "X-PhotonIQ-CustomerID",
+            "in": "header",
+            "description": "The customer ID",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "requestBody": {
+          "content": {
+            "multipart/form-data": {
+              "schema": {
+                "$ref": "#/components/schemas/FunctionDataForm"
+              }
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully deployed the function."
+          },
+          "404": {
```

Comment:
@dlozina-macrometa, @felipe-lopes-macrometa, @joubert-macrometa  - In each endpoint we have the next successful response:
```
{
  "code": 200,
  "error": false,
  "result": "Successfully deployed the function."
}
```
or failure:
```
{
  "code": 404,
  "error": true,
  "errorMessage": "Customer qaaaa not found"
}
```

Would be good to provide the next schema for `404`:
```
    "404": {
            "description": "Requests count from this IP exceeds the rate limit.",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "code": {
                      "description": "HTTP response code.",
                      "type": "integer",
                      "example": 429
                    },
                    "error": {
                      "description": "Flag if there is an error in response. *True* for this response.",
                      "type": "boolean",
                      "example": true
                    },
                    "errorMessage": {
                      "description": "A message created for this error.",
                      "type": "string",
                      "example": "Error message"
                    }
                  }
                }
              }
            }
  ```          

Note: `execute` method doesn't have response schema  for successful result because payload creates by a function.

Reply:
@dlazarenko-macrometa here you have it as a "404" code and the example is "429".  Usually rate limit errors are denoted by 429.